### DISCUSSION TITLE
Support for Flexible URL Configuration with Cloudflare AI Gateway

### DISCUSSION THREAD

**Discussion by Chinayuesir** [RageAgainstThePixel/com.openai.unity#214] [Q&A] [answered]
Hello,

I am seeking support for integrating with the Cloudflare AI gateway service, which involves using their proxy URL: https://gateway.ai.cloudflare.com/v1/xxx/xxx/openai. When this URL is passed into the proxy domain, due to the underlying mechanism of the OpenAISettings class, the URL gets concatenated to https://gateway.ai.cloudflare.com/v1/xxx/xxx/openai/v1/chat/xxxx. As can be seen, the API version gets duplicated, resulting in an incorrect URL.

I am reaching out to inquire if there's a way to flexibly modify this behavior to accommodate such services without the URL path components being redundantly appended. This would greatly enhance the flexibility and usability of the service integration.

Thank you for considering this improvement.

**Comment by StephenHodgson** [comment]
What are the steps to configure the cloudflare proxy?

**Reply by Chinayuesir** [reply]
https://developers.cloudflare.com/ai-gateway/providers/openai/
According to their doc,it seems that we can just change the url to "https://gateway.ai.cloudflare.com/v1/ACCOUNT_TAG/GATEWAY/openai/chat/completions",but the api version is followed by their main domin.

**Comment by Chinayuesir** [comment, marked as answer]
I downloaded the source code and tried to solve the problem, using a hard-coded form.
In the constructor on line 33 of the file OpenAISettingsInfo.cs, the following code snippet was added:
```
if (ResourceName.Contains("cloudflare"))
{
    BaseRequest = "/openai/";
}
else
{
    BaseRequest = $"/{ApiVersion}/"; 
}
```
Fortunately, it seems to have solved my problem.
These idiosyncratic issues may occur in a very small number of developers; after all, AI Gateway is a beta feature offered by Cloudflare. Downloading the source code and reconfiguring the environment took a bit more time, but it seems to have solved my immediate problem. I may try to modify the other code and apply it to my personal open source project, which is also open source on my github page and is available to many developers who are experiencing similar issues as I am!

**Reply by StephenHodgson** [reply]
I'll look into adding support for cloudflare, since they have proxy services for a few other projects I'm working on as well.
